packages/core/src/errors.ts

```diff
@@ -62,9 +63,35 @@ export class RetryableError extends CustomError {
  */
 export class InvalidAuthenticationError extends CustomError {
   status = 401
-  code = 'invalid_authentication'
+  code = ErrorCodes.INVALID_AUTHENTICATION
```

Comment:
We could do something like the below. This would allow us to continue using the `InvalidAuthenticationError` class for all of our authentication errors. This approach also makes it clear that all auth-related errors should have a status code of 401.

```suggestion
  code = this.code || ErrorCodes.INVALID_AUTHENTICATION
```